internal/rest/middleware/errhandler.go

```diff
@@ -0,0 +1,30 @@
+package middleware
+
+import (
+	"github.com/cockroachdb/errors"
+
+	ierr "github.com/flexprice/flexprice/internal/errors"
+	"github.com/gin-gonic/gin"
+)
+
+// Middleware for handling errors
+func ErrorHandler() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Next()
+		if len(c.Errors) > 0 {
+			err := c.Errors.Last().Err
+
+			// Create error response
+			errResp := gin.H{
+				"success": false,
+				"error": gin.H{
+					"message": errors.UnwrapAll(err).Error(),
+				},
+			}
```

Comment:
I think we will have to suport something like a display_message field so that frontend can also show business validatio errors if required.
If not then maybe codes so that Frontend can handle the error display logic at their end. How will we accomplish this?

The middleware is solving for error meessgae, chainging and proper HTTP status - so that part looks great.

Reply:
> I think we will have to suport something like a display_message field so that frontend can also show business validatio errors if required
thinking about an approach for this